gemrb/core/Scriptable/Actor.cpp

```diff
@@ -3621,8 +3621,8 @@ bool Actor::HasSpecialDeathReaction(const ieVariable& deadname) const
 {
 	AutoTable tm = gamedata->LoadTable("death");
 	if (!tm) return false;
-	const auto& value = tm->QueryField(scriptName.CString(), deadname);
-	return value[0] != '0';
+	const std::string& value = tm->QueryField(scriptName, deadname);
+	return !value.empty();
```

Comment:
This looks wrong, why would it ever return an empty string?

Reply:
are you asking about `HasSpecialDeathReaction` or `QueryField`? the latter returns an empty string for non existent entries. but yeah I mistook `0` for `\0`

Reply:
oh, I guess it returns the default, which in this case is '0'